### PR TITLE
Wrap badrequest error when forking

### LINKED ISSUE #423: Wrap BadRequest error logs when forking
When defaulting to the forked network, if the requested object is missing (e.g. an undeclared class), the feeder gateway client logs the error (without actually raising) - this is not a bug, but may be misleading.

This should be handled in origin.py; we could use the same mechanism used in ForkedState (and apply it in other places in ForkedState where missing)

### PR DESCRIPTION
**Pull request by tonypony220**
## Usage related changes
- Closes #423

Not wrapped functions in Forked state seems to have no errors and not have to be suppressed.

- none

## Development related changes

- redirect stderr replaced with muting only logger (not redirects in origin for some reason). 

## Checklist:

- [x] Applied formatting - `./scripts/format.sh`
- [x] No linter errors - `./scripts/lint.sh`
- [x] Performed code self-review
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [ ] Documented the changes
- [x] Linked the issues which this PR resolves
- [x] Updated the tests
- [ ] All tests are passing - `./scripts/test.sh`

### CODE REVIEW

File: starknet_devnet/util.py [hunk 1]
```diff
@@ -243,3 +244,26 @@ def get_fee_estimation_info(tx_fee: int, gas_price: int):
...
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.logger.disabled = False
+
+
+client_spec = importlib.util.find_spec("services.external_api.client")
+assert client_spec
+
+# FeederGatewayClient is implemented in such a way that it logs and raises;
+# this suppresses the logging
+suppress_feeder_gateaway_client_logger = SuppressLogger(
```
Comment (FabijanC) [inline starknet_devnet/util.py:267]: What's the difference between this and `contextlib.redirect_stderr(None)` that has been used so far?

File: test/test_endpoints.py
Comment (FabijanC) [inline test/test_endpoints.py:1]: Why was it necessary to update the tests?

File: starknet_devnet/util.py (hunk 1 above)
Comment (tonypony220) [reply starknet_devnet/util.py:267]: `contextlib.redirect_stderr(None)` somehow not works when used in ForkedOrigin methods.

File: test/test_endpoints.py
Comment (tonypony220) [reply test/test_endpoints.py:1]: Not necessarily in general, just convenient to check. i will then discard the changes checking the wrapper

File: starknet_devnet/util.py [hunk 2]
```diff
@@ -243,3 +244,26 @@ def get_fee_estimation_info(tx_fee: int, gas_price: int):
...
+        self.logger = logger
+
+    def __enter__(self):
+        self.logger.disabled = True
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.logger.disabled = False
+
+
+client_spec = importlib.util.find_spec("services.external_api.client")
```
Comment (FabijanC) [inline starknet_devnet/util.py:262]: Why is this imported like this?
What about
```python
import services.external_api.client as client_spec
```

And what even is this `services` module? I see it's not native to Python, but it's also not a dependency listed in `pyproject.toml`.

File: starknet_devnet/util.py [hunk 3]
```diff
@@ -243,3 +244,26 @@ def get_fee_estimation_info(tx_fee: int, gas_price: int):
 def warn(msg: str, file=sys.stderr):
     """Log a warning"""
     print(f"\033[93m{msg}\033[0m", file=file)
+
+
+class SuppressLogger:
+    """ctx mng to suppress logger"""
```
Comment (FabijanC) [inline starknet_devnet/util.py:250]: Can you use full words here? Is that supposed to be `Context manager`?

File: starknet_devnet/util.py [hunk 4]
```diff
@@ -243,3 +244,26 @@ def get_fee_estimation_info(tx_fee: int, gas_price: int):
 def warn(msg: str, file=sys.stderr):
     """Log a warning"""
     print(f"\033[93m{msg}\033[0m", file=file)
+
+
+class SuppressLogger:
```
Comment (FabijanC) [inline starknet_devnet/util.py:249]: Class names shouldn't be verbs but nouns (with possible descriptors). E.g. `SuppressedLogger` or `LogSuppressor`.

**Review by FabijanC — CHANGES_REQUESTED** [review]

File: starknet_devnet/util.py (hunk 2 above)
Comment (tonypony220) [reply starknet_devnet/util.py]: Yes, my bad. Rewrote to check the existence of the target logger, not the module. 
The `services` module is associated with Everest, from which FeederGatewayClient is inherited.

File: starknet_devnet/util.py (hunk 3 above)
Comment (tonypony220) [reply starknet_devnet/util.py]: fixed

**Review by FabijanC — APPROVED** [review]

**Pull request merged**